airflow/models/param.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any
+
+from jsonschema import Draft4Validator, FormatChecker
+from jsonschema.exceptions import ValidationError
+
+
+class Param:
```

Comment:
Hmmm, having airflow.models.params.Param and airflow.models.dagparam.DagParam is confusing -- do we still need both?

Probably, as they do different things, but the naming is going to lead to confision 🤔

Reply:
As discussed, moved DagParam to Param as well and added comments.